src/routes/routes.js

```diff
@@ -1,26 +1,33 @@
+import { signin, signup } from '../controllers/authController';
 import {
-  validateSigninFormData, validUser, validateCommentData, validateInvite, validateUUID
+  validateSigninFormData,
+  validateSignupFormData,
+  validUser,
+  validateCommentData,
+  validateInvite,
+  validateUUID
 } from '../middlewares/middlewares';
-import {
-  signin
-} from '../controllers/authController';
-import {
-  getComments, createComment
-} from '../controllers/commentController';
+
+import { getComments, createComment } from '../controllers/commentController';
 import { authenticateUserToken } from '../middlewares/authentication';
 import { upvoteInvite } from '../controllers/upvoteController';
 
 export const initRoutes = app => {
   app.get('/', (req, res) => res.status(200).json({ message: 'Welcome' }));
-  app.get('/post',(req, res) => res.render('userPost'));
+  app.get('/post', (req, res) => res.render('userPost'));
 
   app.post('/api/v1/auth/signin', validateSigninFormData, validUser, signin);
 
+  app.post('/api/v1/auth/signup', validateSignupFormData, signup);
```

Comment:
In your route there should be a middleware that validates if the user has already signed up with that email

`app.post('/api/v1/auth/signup', validateSignupFormData, validateExistingUser, signup);`

`validateExistingUser` should return a 409 conflict if user is already sign up with same email or proceed with next() otherwise.

Reply:
Middleware added and appropriate test passes =>
![image](https://user-images.githubusercontent.com/34924520/67475116-7f4a4800-f64d-11e9-9188-1e6f63d400fd.png)